presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java

```diff
@@ -605,7 +614,12 @@ protected void invalidateTable(String databaseName, String tableName)
     @Override
     public Optional<Partition> getPartition(MetastoreContext metastoreContext, String databaseName, String tableName, List<String> partitionValues)
     {
-        return get(partitionCache, getCachingKey(metastoreContext, hivePartitionName(databaseName, tableName, partitionValues)));
+        KeyAndContext<HivePartitionName> key = getCachingKey(metastoreContext, hivePartitionName(databaseName, tableName, partitionValues));
+        Optional<Partition> result = get(partitionCache, key);
```

Comment:
It seems that it is possible we are comparing metastore against metastore in this case, which is likely fine but just to call it out